File: dataset.R
library(data.table)
library(readxl)
ASD <- fread("data/MRD274/ASD.CSV")
DSP <- fread("data/MRD274/DSP.CSV")
FOS <- fread("data/MRD274/FOS.CSV")


ivar <- "Wavelength_(nm)"
mvars <- colnames(ASD)[!colnames(ASD) %in% ivar]
melt(ASD, id.vars = ivar, measure.vars = mvars, variable.name = "Sample", value.name = "Reflectance") -> ASD


mvars <- colnames(DSP)[!colnames(DSP) %in% ivar]
melt(DSP, id.vars = ivar, measure.vars = mvars, variable.name = "Sample", value.name = "Reflectance") -> DSP

mvars <- colnames(FOS)[!colnames(FOS) %in% ivar]
melt(FOS, id.vars = ivar, measure.vars = mvars, variable.name = "Sample", value.name = "Reflectance") -> FOS

DATA <- rbindlist(list(
  data.table(ID="asd", ASD),
  data.table(ID="dsp", DSP),
  data.table(ID="fos", FOS)
  
))

DATA[,SampleID:=sub(".*:(\\w+).*", "\\1", Sample)]
DATA[,Sample:=NULL]

SampleID <- unique(DATA$SampleID)
LTHA <- fread("data/MRD274/data_A.csv")
LTHB <- fread("data/MRD274/data_B.csv")


IDX <- data.table(SampleID=SampleID, ID=1:length(SampleID),LTHA=SampleID %in% LTHA$SampleID,LTHB=SampleID %in% LTHB$SampleID)
fwrite(IDX, "data/MRD274/IDX.csv")

IDX[LTHA==TRUE & LTHB==TRUE]

File: dataset.py


import pandas as pd
# MRD274 data
ASD1 = pd.read_csv("spectra/dataset/MRD274/ASD.CSV")
DSP = pd.read_csv("spectra/dataset/MRD274/DSP.CSV")
FOS = pd.read_csv("spectra/dataset/MRD274/FOS.CSV")

# BalyaNorth data
ASD2 = pd.read_csv("spectra/dataset/BalyaNorth/ASD.CSV")

# Lithology
LITH_A = pd.read_csv("data/litho_A.csv")
LITH_B = pd.read_csv("data/litho_B_V2.csv")

ivar = "Wavelength_(nm)"
mvars = ASD1.columns.difference([ivar])
ASD1 = pd.melt(ASD1, id_vars=ivar, value_vars=mvars, var_name="Sample", value_name="Reflectance")

mvars = ASD2.columns.difference([ivar])
ASD2 = pd.melt(ASD2, id_vars=ivar, value_vars=mvars, var_name="Sample", value_name="Reflectance")

mvars = DSP.columns.difference([ivar])
DSP = pd.melt(DSP, id_vars=ivar, value_vars=mvars, var_name="Sample", value_name="Reflectance")

mvars = FOS.columns.difference([ivar])
FOS = pd.melt(FOS, id_vars=ivar, value_vars=mvars, var_name="Sample", value_name="Reflectance")

DATA = pd.concat([
    ASD1.assign(SourceID="asd", ProjectID="mrd"),
    ASD2.assign(SourceID="asd", ProjectID="bn"),
    DSP.assign(SourceID="dsp", ProjectID="mrd"),
    FOS.assign(SourceID="fos", ProjectID="mrd"),
], ignore_index=True)

DATA['SampleID'] = DATA['Sample'].str.extract(r'.*:(\w+).*')[0]
DATA = DATA.drop(columns='Sample')

IDX = DATA[['SampleID', 'SourceID', 'ProjectID']].drop_duplicates()

IDX['LithoGeochem_A'] = IDX['SampleID'].isin(LITH_A['SampleID']) 
IDX['LithoGeochem_B'] = IDX['SampleID'].isin(LITH_B['SampleID'])

IDX.to_csv("data/IDX.csv", index=False)
